detection_rules/integrations.py

```diff
@@ -187,54 +190,72 @@ def find_latest_compatible_version(package: str, integration: str,
+                                                       key=lambda x: Version.parse(x)))
 
-        if Version(highest_compatible_version) > Version(rule_stack_version):
+        if highest_compatible_version > rule_stack_version:
             # generate notice message that a later integration version is available
             integration = f" {integration.strip()}" if integration else ""
 
             notice = (f"There is a new integration {package}{integration} version {version} available!",
                       f"Update the rule min_stack version from {rule_stack_version} to "
                       f"{highest_compatible_version} if using new features in this latest version.")
 
-        if int(highest_compatible_version[0]) == int(rule_stack_version[0]):
+        if highest_compatible_version.major == rule_stack_version.major:
             return version, notice
 
         else:
             # Check for rules that cross majors
             for compatible_version in compatible_versions:
-                if Version(compatible_version) <= Version(rule_stack_version):
+                if Version.parse(compatible_version) <= rule_stack_version:
                     return version, notice
 
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
-def get_integration_manifests(integration: str) -> list:
+def get_integration_manifests(integration: str, prerelease: str, kibana_version: str) -> list:
     """Iterates over specified integrations from package-storage and combines manifests per version."""
     epr_search_url = "https://epr.elastic.co/search"
+    if not prerelease:
+        prerelease = "false"
 
     # link for search parameters - https://github.com/elastic/package-registry
-    epr_search_parameters = {"package": f"{integration}", "prerelease": "false",
+    epr_search_parameters = {"package": f"{integration}", "prerelease": prerelease,
                              "all": "true", "include_policy_templates": "true"}
+    if kibana_version:
+        epr_search_parameters["kibana.version"] = kibana_version
     epr_search_response = requests.get(epr_search_url, params=epr_search_parameters)
     epr_search_response.raise_for_status()
     manifests = epr_search_response.json()
 
     if not manifests:
         raise ValueError(f"EPR search for {integration} integration package returned empty list")
 
+    sorted_manifests = sorted(manifests, key=lambda p: Version.parse(p["version"]), reverse=True)
     print(f"loaded {integration} manifests from the following package versions: "
-          f"{[manifest['version'] for manifest in manifests]}")
+          f"{[manifest['version'] for manifest in sorted_manifests]}")
     return manifests
 
 
+def find_latest_integration_version(integration: str, maturity: str, stack_version: Version):
+    """Finds the latest integration version based on maturity and stack version"""
+    prerelease = "false" if maturity == "ga" else "true"
```

Comment:
You can probably do this check in `get_integration_manifests` since you're not using `prerelease` in this method.

Reply:
Sounds good. Timeout shouldn't take over 10 seconds to return unless you have another number in mind.

Reply:
`bump-pkg-versions` in `devtools.py` is the only command that will take maturity values of `ga` or `beta` as values and supply them here. Otherwise, prerelease is either true or false as a parameter for the EPR requests and since the `get_integration_manifests` function may have other references, we should keep `prerelease` and limit to false or true.